WaterLog_Backend/Procedures.cs

```diff
@@ -346,28 +349,38 @@ private string BuildUrl(int segmentId)
             return "https://iot.retrotest.co.za/alert/segment/" + segmentId;
         }
 
-        public double CalculatePerHourWastageCost(SegmentLeaksEntry leak)
+        public async Task<double> CalculatePerHourWastageLitre(SegmentLeaksEntry leak)
         {
-            var list = _db.SegmentEvents;
-            var entry = list
-            .Where(inlist => inlist.SegmentsId == leak.SegmentsId)
-            .Last();
+            var list = await _db.SegmentEvents.Where(inlist => inlist.SegmentsId == leak.SegmentsId && inlist.EventType == "leak" && inlist.TimeStamp >= leak.OriginalTimeStamp && inlist.TimeStamp <= leak.LatestTimeStamp).ToListAsync();
 
-            double currentTariff = Globals.RandPerLitre;
-            double usageperpoll = (entry.FlowIn - entry.FlowOut);
-            var totalPH = (usageperpoll * currentTariff);
-            return Math.Max(totalPH, 0);
+            double usageperpoll = 0.0;
+            foreach (var item in list)
+            {
+                usageperpoll += (item.FlowIn - item.FlowOut);
+            }
+            usageperpoll *= 0.0167;
+            var minutes = (leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalMinutes;
+            usageperpoll /= minutes;
+            usageperpoll *= 60.00;
+            return Math.Max(usageperpoll,0);
         }
 
-        public double CalculatePerHourWastageLitre(SegmentLeaksEntry leak)
+        public async Task<double> CalculatePerHourWastageCost(SegmentLeaksEntry leak)
         {
-            var list = _db.SegmentEvents;
-            var entry = list
-            .Where(inlist => inlist.SegmentsId == leak.SegmentsId)
-            .Last();
+            var list = await _db.SegmentEvents.Where(inlist => inlist.SegmentsId == leak.SegmentsId && inlist.EventType == "leak" && inlist.TimeStamp >= leak.OriginalTimeStamp && inlist.TimeStamp <= leak.LatestTimeStamp).ToListAsync();
+
+            double usageperpoll = 0.0;
+            foreach(var item in list)
+            {
+                usageperpoll += (item.FlowIn - item.FlowOut);
+            }
+            //Convert Poll Into Minutes
+            usageperpoll *= 0.0167;
+            var minutes = (leak.LatestTimeStamp - leak.OriginalTimeStamp).TotalMinutes;
+            usageperpoll /= minutes;
+            usageperpoll *= 60.00;
```

Comment:
Again 60 being used repeatedly. Make use of descriptive constants

Reply:
Moved to Globals class